src/realm/sync/noinst/client_reset_recovery.cpp

```diff
@@ -547,14 +547,55 @@ bool RecoverLocalChangesetsHandler::resolve_path(ListPath& path, Obj remote_obj,
                 return false;
             }
         }
-        else { // single link to embedded object
-            // Neither embedded object sets nor Mixed(TypedLink) to embedded objects are supported.
-            REALM_ASSERT_EX(!col.is_collection(), col);
-            REALM_ASSERT_EX(col.get_type() == col_type_Link, col);
-            StringData col_name = remote_obj.get_table()->get_column_name(col);
-            remote_obj = remote_obj.get_linked_object(col);
-            local_obj = local_obj.get_linked_object(col_name);
-            ++it;
+        else {
```

Comment:
you could remove one level of indentation by using `else if ( ... == col_type_Mixed) ... {`